snuba/snuba_migrations/search_issues/0001_search_issues.py

```diff
@@ -0,0 +1,96 @@
+    Column("fingerprint", Array(String())),
+    Column("occurrence_id", UUID()),
+    Column("occurrence_type_id", UInt(8)),
+    Column("detection_timestamp", DateTime()),
+    Column("event_id", UUID(Modifiers(nullable=True))),
+    Column("trace_id", UUID(Modifiers(nullable=True))),
+    Column("platform", String(Modifiers(low_cardinality=True))),
+    Column("environment", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("release", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("dist", String(Modifiers(low_cardinality=True, nullable=True))),
+    Column("receive_timestamp", DateTime()),
+    Column("client_timestamp", DateTime()),
+    Column("tags", Nested([("key", String()), ("value", String())])),
+    Column("user", String(Modifiers(default="''"))),
+    Column("user_hash", UInt(64, Modifiers(materialized="cityHash64(user)"))),
+    Column("user_id", String(Modifiers(nullable=True))),
+    Column("user_name", String(Modifiers(nullable=True))),
+    Column("user_email", String(Modifiers(nullable=True))),
+    Column("ip_address_v4", IPv4(Modifiers(nullable=True))),
+    Column("ip_address_v6", IPv6(Modifiers(nullable=True))),
+    Column("sdk_name", String(Modifiers(low_cardinality=True, default="''"))),
+    Column("sdk_version", String(Modifiers(low_cardinality=True, default="''"))),
+    Column("contexts", Nested([("key", String()), ("value", String())])),
+    Column("http_method", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("http_referer", String(Modifiers(nullable=True))),
+    Column("retention_days", UInt(16)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.SEARCH_ISSUES,
+                table_name="search_issues_local",
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    order_by="(organization_id, project_id, toStartOfDay(detection_timestamp))",
+                    partition_by="(retention_days, toMonday(detection_timestamp))",
+                    # TODO: need to figure out what sample is here
+                    # sample_by="cityHash64(occurrence_id)",
+                    settings={"index_granularity": "8192"},
+                    storage_set=StorageSetKey.SEARCH_ISSUES,
+                    ttl="detection_timestamp + toIntervalDay(retention_days)",
+                ),
+                target=OperationTarget.LOCAL,
+            ),
+            operations.CreateTable(
+                storage_set=StorageSetKey.SEARCH_ISSUES,
+                table_name="search_issues_dist",
+                columns=columns,
+                engine=table_engines.Distributed(
+                    local_table_name="search_issues_local",
+                    sharding_key="cityHash64(project_id)",
```

Comment:
Not sure what the best sharding key is for this.

Reply:
Historically, we have sharded the data so that it gets spread evenly across all shards of the cluster. I don't think there is a reason to deviate from that at the moment. So you could use `cityHash64(occurence_id)` here

Reply:
For distributed aggregations, does each shard have to have all the rows in the group by clause to work effectively or is there some mechanism that would aggregate across shards?

Ex. 
search_issues_dist Shard 1:
------------------------------
occurrence_1, group_1
occurrence_2, group_2
occurrence_3, group_3
occurrence_4, group_4

search_issues_dist Shard 2:
------------------------------
occurrence_5, group_2
occurrence_6, group_2
occurrence_7, group_5
occurrence_8, group_6

query:
```
select group_id, count() as count 
from search_issues_dist
where group_id=group2
group by group_id
```

I think we mostly do aggregations by group_id for this dataset - would sharding on group_id make sense here?

Reply:
Would it make sense to shard on `project_id`, that way itll support aggregations on project and group between shards - assuming there is a limitation on aggregations across shards.